JWT를 이용한 로그인 구현

jgm0327 opened the pull request:
- jwt를 위한 dependency 추가
- exception 추가
- repository에 email로 사용자 조회하는 메소드 추가
- controller와 service에 로그인을 위한 로직 추가
- 로그인을 security 설정 수정

jgm0327 committed:
Feat: jwt를 위한 설정

- dependency 추가와 secret key 설정

Resolves: #5

jgm0327 committed:
Feat: 에러 처리 추가

- GlobalExceptionHandler 클래스에서 DanggunMarketException으로 모든 예외 처리할 수 있도록 수정

- 비밀번호 다를 때 에러 처리를 위해 MemberErrorCode와 MemberNotFoundException 추가

- 회원 정보가 없을 때 에러 처리를 위해 MemberErrorCode와 WrongPasswordException 추가

Resolves: #5

jgm0327 committed:
Feat: 에러 처리 추가

- role 추가를 위해 enum 추가 후 MemberEntity에 추가

Resolves: #5

jgm0327 committed:
Feat: jwt filter를 security에 등록

- jwt를 위한 JwtUtil과 jwt 확인하기 위한 JwtFilter 추가

- email로 유저를 찾을 수 있도록 MemberRepository 메소드 추가

- UserDetails를 상속하는 LoggedInMember 추가

- SecurityConfig에 filter 추가 및 /v1/login 경로에는 시큐리티 안 타도록 함

Resolves: #5

jgm0327 committed:
Feat: 로그인을 위한 controller와 service 수정

- controller에서 jwt를 헤더에 넣어서 응답을 보내도록 구현 및 닉네임과 이메일을 바디에 넣어서 응답을 보냄

- service에서 로그인을 위한 로직 구현

Resolves: #5

f-lab-kane commented on danggunMarket/src/main/java/com/example/danggunmarket/member/MemberService.java:
```diff
@@ -24,15 +27,33 @@ public void join(JoinRequest joinRequest){
+    public LoginResponse login(LoginRequest loginRequest) {
+        String email = loginRequest.getEmail();
+
+        System.out.println(email);
+
+        System.out.println(memberRepository.existsByEmail(email));
```
SLF4J나 Logback 을 활용하는것지 좋을거 같아요.

jgm0327 replied:
저번에 말씀드렸던 에러 잡는 것때문에 임시로 작성했던 것이 올라갔네요.

저거는 지우고 로그 처리할 때는 logback 사용하도록 하겠습니다.

f-lab-kane reviewed (approved):
규민님 수고하셨어요.
다른 api에서 로그인 정보가 필요할땐 AuthArgumentResolver를 활용해보는것도 좋습니다.

참고) https://velog.io/@uiurihappy/Spring-Argument-Resolver-%EC%A0%81%EC%9A%A9%ED%95%98%EC%97%AC-%EC%9C%A0%EC%A0%80-%EC%A0%95%EB%B3%B4-%EB%B6%88%EB%9F%AC%EC%98%A4%EA%B8%B0

jgm0327 committed:
Feat: 로그인을 위한 controller와 service 수정

- controller에서 jwt를 헤더에 넣어서 응답을 보내도록 구현 및 닉네임과 이메일을 바디에 넣어서 응답을 보냄

- service에서 로그인을 위한 로직 구현

Resolves: #5
